ament_pep257/ament_pep257/main.py

```diff
@@ -107,6 +107,10 @@ def main(argv=sys.argv[1:]):
     return rc
 
 
+def _filename_in_excludes(filename, excludes):
+    return any([os.path.abspath(filename).startswith(e) for e in excludes])
```

Comment:
You can't use `startswith` like this to check if the `filename` is in a list of directories.

Counter example: `exclude: /foo`, `filename: /foobar` which would incorrectly return `True`.

Reply:
Good point, thank you.
Now checking if `filename` is exact `exclude` or is prefixed with `exclude/`.

Reply:
This won't work in Windows which uses backslashes.

Reply:
Thanks, completely forgot about that.
The newest version got rid of `startwith` in favor of `commonpath`.

Reply:
`commonpath` will have the same problem. The easiest way to do the check is something like this:

```
path = pathlib.Path(filename)
exclude = pathlib.Path(exclude)
exclude == p or exclude in path.parents
```

Reply:
> `commonpath` will have the same problem

Tested on Windows 10, Python 3.6.8:
```python
>>> os.path.abspath('1.txt')
'C:\\test\\1.txt'
>>> f = "C:\\test\\1.txt"
>>> f1 = "C:\\test2\\1.txt" 
>>> e = "C:\\test"
>>> os.path.commonpath([f,e]) == e
True
>>> os.path.commonpath([f1,e]) == e 
False
```

Am I missed something?

Reply:
Oh, I confused it with `commonprefix`. `commonpath` is good as is.